src/app/vault/ciphers.component.ts

```diff
@@ -81,6 +89,79 @@ export class CiphersComponent extends BaseCiphersComponent implements OnDestroy
     this.didScroll = this.pagedCiphers.length > this.pageSize;
   }
 
+  async loadOrganizationNames(): Promise<Map<string, string>> {
+    const organizations = await this.organizationService.getAll();
+    const organizationNames = new Map<string, string>();
+
+    organizations.forEach((organization) => {
+      organizationNames.set(organization.id, organization.name);
+    });
+
+    return organizationNames;
+  }
+
+  getOrganizationById(c: CipherView): string {
+    return c.organizationId != null
+      ? this.organizationNames.get(c.organizationId)
+      : this.i18nService.t("myVault");
+  }
+
+  sortCiphers(sortBy: string): void {
+    if (sortBy === this.sortBy) {
+      this.sortedDescending = !this.sortedDescending;
+    } else {
+      this.sortBy = sortBy;
+      this.sortedDescending = true;
+    }
+
+    if (sortBy === "lastEdited") {
+      if (this.sortedDescending) {
+        this.ciphers.sort((a: CipherView, b: CipherView) => {
+          return b.revisionDate.valueOf() - a.revisionDate.valueOf();
+        });
+      } else {
+        this.ciphers.sort((a: CipherView, b: CipherView) => {
+          return a.revisionDate.valueOf() - b.revisionDate.valueOf();
+        });
+      }
+      this.showData = "lastEdited";
+    } else if (sortBy === "dateCreated") {
+      // TO-DO
+    } else if (sortBy === "owner") {
+      this.ciphers.sort((a: CipherView, b: CipherView) => {
+        const sortingValue1 =
+          a.organizationId == null ? "Personal" : this.organizationNames.get(a.organizationId);
+        const sortingValue2 =
+          b.organizationId == null ? "Personal" : this.organizationNames.get(b.organizationId);
+
+        if (this.sortedDescending) {
+          return sortingValue1.localeCompare(sortingValue2);
+        } else {
+          return sortingValue2.localeCompare(sortingValue1);
+        }
+      });
+      this.showData = "owner";
+    } else if (sortBy === "name") {
+      if (this.sortedDescending) {
+        this.ciphers.sort((a, b) => a.name.localeCompare(b.name));
+      } else {
+        this.ciphers.sort((a, b) => b.name.localeCompare(a.name));
+      }
+    }
+  }
+
+  reverseSort(): void {
+    this.sortCiphers(this.sortBy);
+  }
+
+  changeDataToShow(type: string): void {
+    this.showData = type;
+  }
+
+  revisionDate(date: Date): string {
+    return date ? date.toLocaleDateString("en-US") : null;
```

Comment:
We should get the user's locale from `i18nService` and use that instead of assuming they want it in en-US format.

Style: we use explicit null checks (`date != null`).